pom.client.xml

```diff
@@ -550,17 +538,84 @@
 
    <profiles>
 
+       <!--
+        By default we build against our baseline, Java 8, but we also want to ensure compatibility
+        against the latest Java LTS release, currently Java 11. We therefore have two profiles:
+          * The default 'java8', which will perform a build using Java 8 as its target.
+          * The non-default 'java-lts' profile, which will build against the current LTS release.
+      -->
       <profile>
-
-         <id>run-shade-plugin</id>
+         <id>java8</id>
          <activation>
-            <property><name>runShadePlugin</name></property>
+            <activeByDefault>true</activeByDefault>
```

Comment:
Did some testing and this doesn't appear to work. The profile is not activated because of the following behavior regarding activeByDefault from Maven's docs:

> This profile will automatically be active for all builds unless another profile in the same pom is activated using one of the previously described methods. All profiles that are active by default are automatically deactivated when a profile in the pom is activated on the command line or through its activation config.

Because the javadoc-no-modules-fix profile is being activated by the use of JDK 9+, it's causing the "default" profile not to activate.

The workaround suggested seems to be to use a property to switch between the two profiles, rather than naming the profile on the CLI. Basically, we'd add a condition to activate the "java8" profile when the property is absent, and another to activate the "java-lts" profile when it's present. We'd then pass `-Djava-lts` to build with the java-lts profile (and omit it to build with the java8 profile) rather than using `-P`.

```diff
diff --git a/pom.client.xml b/pom.client.xml
index 38d0172cf0..9613323412 100644
--- a/pom.client.xml
+++ b/pom.client.xml
@@ -548,6 +548,9 @@
          <id>java8</id>
          <activation>
             <activeByDefault>true</activeByDefault>
+            <property>
+                <name>!java-lts</name>
+            </property>
          </activation>
          <build>
             <plugins>
@@ -569,6 +572,11 @@

       <profile>
          <id>java-lts</id>
+         <activation>
+            <property>
+                <name>java-lts</name>
+            </property>
+        </activation>
          <build>
             <plugins>
                <plugin>
```

Reply:
BTW I was using the following command to see what profiles were active in a given situation: `mvn help:active-profiles -f pom.client.xml`